implementations/rust/ockam/ockam_command/src/status.rs

```diff
@@ -0,0 +1,125 @@
+    status: String,
+}
+
+impl StatusCommand {
+    pub fn run(self, options: CommandGlobalOpts) {
+        node_rpc(rpc, (options, self));
+    }
+}
+
+async fn rpc(ctx: Context, (opts, cmd): (CommandGlobalOpts, StatusCommand)) -> Result<()> {
+    run_impl(&ctx, opts, cmd).await
+}
+
+async fn run_impl(ctx: &Context, opts: CommandGlobalOpts, cmd: StatusCommand) -> Result<()> {
+    let node_states = match opts.state.nodes.list() {
+        Ok(nodes) => nodes,
+        Err(_err) => {
+            return Err(crate::Error::new(
+                exitcode::IOERR,
+                anyhow!("No nodes registered on this system!"),
+            ));
+        }
+    };
+
+    let mut node_details: Vec<NodeDetails> = vec![];
+    for node_state in &node_states {
+        let node_infos = NodeDetails {
+            identity: node_state.config.identity(ctx).await?,
+            state: node_state.clone(),
+            status: get_node_status(ctx, &opts, node_state).await?,
+        };
+        node_details.push(node_infos);
+    }
+
+    let mut status_identities: Vec<IdentityState> = vec![];
+    for identity in opts.state.identities.list()? {
+        if cmd.all {
+            status_identities.push(identity)
+        } else {
+            match &identity.config.enrollment_status {
+                Some(_enrollment) => status_identities.push(identity),
+                None => (),
+            }
+        }
+    }
+
+    print_status(&opts, status_identities, node_details).await?;
+
+    Ok(())
+}
+
+async fn get_node_status(
+    ctx: &Context,
+    opts: &CommandGlobalOpts,
+    node_state: &NodeState,
+) -> Result<String> {
+    let mut node_status: String = "Stopped".to_string();
+
+    let tcp = TcpTransport::create(ctx).await?;
+    let mut rpc = RpcBuilder::new(ctx, opts, &node_state.config.name)
+        .tcp(&tcp)?
+        .build();
+    if rpc
+        .request_with_timeout(api::query_status(), Duration::from_millis(200))
+        .await
+        .is_ok()
+    {
+        let resp = rpc.parse_response::<NodeStatus>()?;
+        node_status = resp.status.to_string();
+    }
+
+    Ok(node_status)
+}
+
+async fn print_status(
+    opts: &CommandGlobalOpts,
+    identities: Vec<IdentityState>,
+    node_details: Vec<NodeDetails>,
+) -> Result<()> {
+    let default_identity = opts.state.identities.default()?;
+
+    for (i_idx, identity) in identities.iter().enumerate() {
+        println!("Identity[{}]", i_idx);
+        if default_identity.config.identifier == identity.config.identifier {
+            println!("{:2}Default: yes", "")
+        }
+        for line in identity.to_string().lines() {
+            println!("{:2}{}", "", line);
+        }
+        if !node_details.is_empty() {
+            println!("{:2}Linked Nodes:", "");
+            for (n_idx, node) in node_details.iter().enumerate() {
+                println!("{:4}Node[{}]:", "", n_idx);
+                if node.identity.identifier() == &identity.config.identifier {
```

Comment:
Note that the two prints done before this line will be printed for every identity, even if only one identity has linked nodes. You should buffer the print contents and print it after you know for sure that the current identity has linked nodes.